pkg/didcomm/crypto/legacy/authcrypt/authcrypt_test.go

```diff
@@ -0,0 +1,198 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package authcrypt
+
+import (
+	"crypto/rand"
+	"encoding/base64"
+	"fmt"
+	"io"
+	"testing"
+
+	"github.com/btcsuite/btcutil/base58"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/crypto/nacl/box"
+	"golang.org/x/crypto/nacl/sign"
+)
+
+func TestEncrypt(t *testing.T) {
+
+	/*
+			Note: box.GenerateKey() generates a curve25519 keypair, not an Ed25519 keypair.
+			I think we should be generating Ed25519 keypairs since they are converted into
+				curve25519 keypairs for processing.
+
+		TODO: use Ed25519 keys, convert to curve25519 internally, encode the Ed25519 form for message packing
+	*/
+
+	t.Run("Success test case: given keys, generate envelope", func(t *testing.T) {
```

Comment:
it would be good to compare the actual result to an expected value

Reply:
Now that I can run the crypter using a deterministic rand source, I have a few tests which do this. Got some ideas on more I can do though.